From d4c272c97b124fab93f6db1e57df1eeff2a59391 Mon Sep 17 00:00:00 2001
From: Marsell Kukuljevic <marsell@joyent.com>
Date: Mon, 25 Feb 2019 15:51:09 +0100
Subject: [PATCH] TRITON-1253 dapi should prevent VMs requiring flexible disk
 sizing being allocated to platforms which are too old to support it Reviewed
 by: Mike Gerdts <mike.gerdts@joyent.com> Approved by: Mike Gerdts
 <mike.gerdts@joyent.com>

---
 docs/index.md      | 1 +
 lib/designation.js | 6 +++++-
 package.json       | 2 +-
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/docs/index.md b/docs/index.md
index 76725d0..07a2880 100644
--- a/docs/index.md
+++ b/docs/index.md
@@ -72,6 +72,7 @@ found in sapi_manifests/cnapi/template.
 | **dapi.changeDefaults**   | Object | -       | This provides some means to override VM allocation behaviour.       |
 | **dapi.changeDefaults.server_spread**        | String | -            | **DEPRECATED** How VMs are spread across CNs (one of: min-ram, max-ram, min-owner, and random)   |
 | **dapi.changeDefaults.filter_docker_min_platform** | String | -      | If present, minimum platform version useful for Docker instances.        |
+| **dapi.changeDefaults.filter_flexible_disk_min_platform** | String | - | If present, minimum platform version useful for instances with flexible disk sizing. |
 | **dapi.changeDefaults.filter_headnode**      | String | true         | Whether VMs cannot allocate on the headnode.                             |
 | **dapi.changeDefaults.filter_min_resources** | String | true         | Whether CPU/RAM/disk limits are ignored when allocating.                 |
 | **dapi.changeDefaults.filter_large_servers** | String | true         | Whether large servers are reserved for larger allocations.               |
diff --git a/lib/designation.js b/lib/designation.js
index 3f7c58a..cfd1121 100644
--- a/lib/designation.js
+++ b/lib/designation.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -39,6 +39,8 @@ var DEFAULT_FILTER_DOCKER_NFS_VOLUMES_AUTOMOUNT_MIN_PLATFORM =
     '20160613T123039Z';
 var DEFAULT_FILTER_NON_DOCKER_NFS_VOLUMES_AUTOMOUNT_MIN_PLATFORM =
     '20170925T211846Z';
+var DEFAULT_FILTER_FLEXIBLE_DISK_MIN_PLATFORM = '20181206T190647Z';
+
 
 function Designation() {}
 
@@ -157,6 +159,8 @@ function getDefaults(changeDefaults) {
         DEFAULT_FILTER_DOCKER_NFS_VOLUMES_AUTOMOUNT_MIN_PLATFORM);
     setDefault('filter_non_docker_nfs_volumes_automount_min_platform',
         DEFAULT_FILTER_NON_DOCKER_NFS_VOLUMES_AUTOMOUNT_MIN_PLATFORM);
+    setDefault('filter_flexible_disk_min_platform',
+        DEFAULT_FILTER_FLEXIBLE_DISK_MIN_PLATFORM);
     setDefault('weight_current_platform', DEFAULT_WEIGHT_CURRENT_PLATFORM);
     setDefault('weight_next_reboot',      DEFAULT_WEIGHT_NEXT_REBOOT);
     setDefault('weight_num_owner_zones',  DEFAULT_WEIGHT_NUM_OWNER_ZONES);
diff --git a/package.json b/package.json
index 07afbbe..174f150 100644
--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
     "assert-plus": "1.0.0",
     "async": "1.5.2",
     "bunyan": "1.8.5",
-    "dapi": "git+https://github.com/joyent/sdc-designation.git#35914310ca9b359ad64b95ab4ea42b763399694a",
+    "dapi": "git+https://github.com/joyent/sdc-designation.git#296a786be98c325c9f30eff4dce16ed04047c9b5",
     "deep-equal": "1.0.1",
     "dox": "0.9.0",
     "gc-stats": "1.2.0",
-- 
2.21.0

